### PR TITLE
adds support for editable origin header

### PR DESCRIPTION
**Pull request by risingjason**
#384

Should be able to resolve above issue.

### CODE REVIEW

**Comment by risingjason** [comment]
I overlooked a test case. Let me resolve real quick

File: src/_fetch/_headers.js [hunk 1]
```diff
@@ -5,16 +5,27 @@ const makeHeaders = function(options) {
...
     headers: {
       'Content-Type': 'application/json',
       'Api-User-Agent': agent,
       'User-Agent': agent,
-      Origin: '*'
+      Origin: origin
     },
     redirect: 'follow'
   }
+  console.log(opts)
```
Comment (wvanderp) [inline src/_fetch/_headers.js:28]: you left a console log here

File: src/_fetch/_headers.js [hunk 2]
```diff
@@ -5,16 +5,27 @@ const makeHeaders = function(options) {
...
+  let origin
+  if (options.noOrigin) {
+    origin = ''
+  } else {
+    origin = 
+      options.origin ||
+      options.Origin ||
+      '*'
+  }
+
```
Comment (wvanderp) [inline src/_fetch/_headers.js:17]: the if statement is fine but a [Conditional (ternary) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)

File: src/_fetch/_headers.js [hunk 3]
```diff
@@ -5,16 +5,27 @@ const makeHeaders = function(options) {
     options['Api-User-Agent'] ||
     'User of the wtf_wikipedia library'
 
+  let origin
+  if (options.noOrigin) {
+    origin = ''
```
Comment (wvanderp) [inline src/_fetch/_headers.js:10]: It will now send emty string as the orgin header. Is that what you want to have happen?

otherwise set it to undefined

**Comment by spencermountain** [comment]
🏅
thank you!

**Comment by risingjason** [comment]
uh oh, i wanted to address @wvanderp 's comments but it got merged down. i can make a new PR for it

**Pull request merged**